ckb-integration-test/src/case/rfc0028/v2019.rs

```diff
@@ -0,0 +1,207 @@
+            make_all_nodes_synced: false,
+            make_all_nodes_connected_and_synced: false,
+            node_options: vec![
+                NodeOptions {
+                    node_name: String::from("node2019"),
+                    ckb_binary: CKB2019.read().unwrap().clone(),
+                    initial_database: "testdata/db/Epoch2V1TestData",
+                    chain_spec: "testdata/spec/ckb2019",
+                    app_config: "testdata/config/ckb2019",
+                },
+                NodeOptions {
+                    node_name: String::from("node2021"),
+                    ckb_binary: CKB2021.read().unwrap().clone(),
+                    initial_database: "testdata/db/Epoch2V2TestData",
+                    chain_spec: "testdata/spec/ckb2021",
+                    app_config: "testdata/config/ckb2021",
+                },
+            ]
+            .into_iter()
+            .collect(),
+        }
+    }
+
+    fn run(&self, nodes: Nodes) {
+        let node2019 = nodes.get_node("node2019");
+        let node2021 = nodes.get_node("node2021");
+
+        node2021.mine(1);
+        node2019
+            .pull_node(node2021)
+            .expect("node2019 pull node2021");
+        let median_time_block_count = node2021.consensus().median_time_block_count.value();
+        let t = node2021.get_tip_block().timestamp();
+        let old_tip_number = node2021.get_tip_block_number();
+
+        // [(relative_millis, input_median_time, input_committed_time, tip_median_time, expected_result)]
+        let cases = vec![
+            (1000, t + 2000, t + 3000, t + 5000, PASS),
+            (1000, t + 2000, t + 3000, t + 4000, PASS),
+            (1000, t + 2000, t + 4000, t + 4000, PASS),
+            (1000, t + 2000, t + 3000, t + 1999, ERROR_IMMATURE),
+        ];
+
+        for (
+            i,
+            (
+                relative_millis,
+                input_median_time,
+                input_committed_time,
+                tip_median_time,
+                expected_result,
+            ),
+        ) in cases.into_iter().enumerate()
+        {
+            assert!(input_median_time < input_committed_time);
+
+            // Use a standalone node to run a case
+            let node = node2021.clone_node(&format!("{}-cloned-{}", node2021.node_name(), i));
+            ckb_testkit::info!(
+                "[Node {}] run case-{}, old_tip_number: {}",
+                node.node_name(),
+                i,
+                old_tip_number
+            );
+
+            let input_committed_number = old_tip_number + median_time_block_count + 1;
+            let new_tip_number = input_committed_number + median_time_block_count;
+            let mut instructions = vec![
+                BuildInstruction::HeaderTimestamp {
+                    block_number: input_committed_number
+                        - (median_time_block_count - median_time_block_count / 2),
+                    timestamp: input_median_time,
```

Comment:
I suggest setting it to a value which does not equal to `input_median_time` in case the implementation use the timestamp as the input median time.